waspc/data/Generator/templates/react-app/src/actions/index.js

```diff
@@ -0,0 +1,74 @@
+import {
+  useMutation,
+  useQueryClient,
+} from 'react-query'
+
+export { configureQueryClient } from '../queryClient'
+
+export function useAction(actionFn, actionOptions) {
+  const queryClient = useQueryClient();
+
+  // TODO(filip): Do we allow all react query options or should we consider a list of allowed options?
+  let options = {}
+
+  if (actionOptions?.optimisticUpdates) {
+    const optimisticUpdatesConfig = makeOptimisticUpdatesConfig(actionOptions.optimisticUpdates)
+    options = makeOptimisticUpdateOptions(queryClient, optimisticUpdatesConfig)
+  }
+
+  return useMutation(actionFn, options)
+}
+
+
+function makeOptimisticUpdatesConfig(optimisticUpdatesConfig) {
+  return optimisticUpdatesConfig.map(({ getQuery, ...rest }) => ({
+    getQuery: (item) => parseQueryKey(getQuery(item)),
+    ...rest,
+  }))
+}
+
+function makeOptimisticUpdateOptions(queryClient, optimisticUpdatesConfig) {
```

Comment:
This entire function is an implementation of [react-query's recommended approach for optimistic updates](https://react-query.tanstack.com/guides/optimistic-updates).

Reply:
We could mention that in the comments of the function.

Reply:
We definitely could, yes :)